Add support for custom log levels

radiophysicist opened the pull request:
It is possible to register custom log levels using `logging.addLevelName()`.
If an event with such a level is to be pushed into sentry, `structlog-sentry` crashes on determining numeric value of the level.

paveldedik commented on structlog_sentry/__init__.py:
```diff
@@ -168,6 +168,26 @@ def _handle_breadcrumb(self, event_dict: EventDict) -> None:
+            return level
+
+        # Log level value not found
+        return default_value
```
I think it would be better to fall back to the original attribute error if it is not a custom log level. Or what do you think?

radiophysicist replied:
Changed to re-raise original exception

paveldedik reviewed (changes requested):
Sorry it took me so long to respond.

Thank you very much for the PR, are you planning on finishing it? I would take over if not. 

BTW please add tests.

paveldedik commented on structlog_sentry/__init__.py:
```diff
@@ -168,6 +168,21 @@ def _handle_breadcrumb(self, event_dict: EventDict) -> None:
+        """Get numeric value for the log level name given."""
+        try:
+            # Try to get one of predefined log levels
+            return getattr(logging, level_name)
```
Wouldn't this fix the issue you described?

```suggestion
            return getattr(logging, level_name.upper())
```

paveldedik replied:
Aha, I see, thank you for explaining it. I think I can approve this and will create a separate issue for this. I just have to think how to fix that.

paveldedik commented on structlog_sentry/__init__.py:
```diff
@@ -168,6 +168,21 @@ def _handle_breadcrumb(self, event_dict: EventDict) -> None:
+                return level
+
+            # Re-raise original error
+            raise
```
I'm thinking that maybe this would be more understandable?

```suggestion
            raise ValueError(f"{level_name} is not a valid log level") from e
```

paveldedik reviewed (commented):
Thank you for the change, I think it is ok now :-) I just have a couple of suggestions that could make it even better :-) Tell me what you think.

paveldedik reviewed (changes requested):
Can you please fix the commit message so that the pipeline passes? :-) https://www.conventionalcommits.org/en/v1.0.0/

radiophysicist commented:
> Can you please fix the commit message so that the pipeline passes? :-) https://www.conventionalcommits.org/en/v1.0.0/

Rebased to master, squashed commits with new message

paveldedik commented:
Ok, merged, I'll release it tomorrow, thank you for the contribution :)
